instrumentation/opentelemetry/github.com/gin-gonic/hypergin/gin_test.go

```diff
@@ -0,0 +1,86 @@
+package hypergin
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/hypertrace/goagent/instrumentation/opentelemetry/internal"
+	sdkhttp "github.com/hypertrace/goagent/sdk/instrumentation/net/http"
+	"go.opentelemetry.io/otel/trace"
+	"gotest.tools/assert"
+)
+
+// inspired in https://github.com/jcchavezs/httptest-php/blob/e6a65c73/src/HttpTest/HttpTestServer.php#L150
+func findAvailablePort() (int, error) {
+	for port := 60000; port < 65535; port++ {
+		l, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+		defer l.Close()
+
+		if err == nil {
+			return port, nil
+		}
+	}
+
+	return 0, errors.New("failed to find an available port")
+}
+
+func TestSpanRecordedCorrectly(t *testing.T) {
```

Comment:
I'd add a test case where we use incoming trace context as this is fundamental (of course we expect otelhttp package to handle it)

Reply:
@jcchavezs would this look like sending in a request with trace headers or something else?

Reply:
Not sure. I don't see the client passing tracing headers to the server.